src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(value, null, null);
+            first = newNode;
+        } else {
+            newNode = new Node<>(value, last, null);
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> before = entry(index);
+            Node<T> eldest = before.prev;
+            Node<T> newNode = new Node<>(value, eldest, before);
+            before.prev = newNode;
+            if (eldest == null) {
+                first = newNode;
+            } else {
+                eldest.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return entry(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = entry(index);
+        T previousValue = node.value;
+        node.value = value;
+        return previousValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> target = entry(index);
+        removed(target);
+        return target.value;
     }
 
     @Override
     public boolean remove(T t) {
+        Node<T> node = first;
+        for (int i = 0; i < size; i++) {
+            if (t == node.value || node.value.equals(t)) {
```

Comment:
what's gonna happen if node.value is null?

Reply:
corrected